### ISSUE TITLE
Items not removed from library.

### ISSUE THREAD

**Issue by torque** [dokibox/dokibox#53]
If an item is removed from a monitored folder while dokibox is not running, it will not be removed when it is started. It won't be removed when trying to add it to a playlist although dokibox will log the following errors:

```
Failure in updateFromFile: (not able to load tags) for [file]
Failure to insertTrackWithFilename: [file]
```

**Comment by mileswu** [comment]
2015-02-08 21:37:35:288 V [main] (Library.m:82) Detected removal of dir at: /Users/mileswu/Music/Ellie Goulding
2015-02-08 21:37:35:292 V [main] (LibraryView.m:846) Back on main thread took 0.000228 sec
2015-02-08 21:37:35:293 V [dokibox.library] (Library.m:347) Deleting file: /Users/mileswu/Music/Ellie Goulding/Ellie Goulding -- Lights (Bassnectar Remix).mp3
2015-02-08 21:37:35.324 dokibox[46863:14424997] CoreData could not fulfill a fault for '0x440002b <x-coredata://8A564CB1-B64E-4F07-AECE-81BFACF96835/artist/p17>'
2015-02-08 21:37:35.353 dokibox[46863:14424997] (
	0   CoreFoundation                      0x00007fff946db66c __exceptionPreprocess + 172
	1   libobjc.A.dylib                     0x00007fff87c0476e objc_exception_throw + 43
	2   CoreData                            0x00007fff8d5cd5c5 _PFFaultHandlerLookupRow + 2213
	3   CoreData                            0x00007fff8d5cc8b1 _PF_FulfillDeferredFault + 305
	4   CoreData                            0x00007fff8d5cc6a4 _sharedIMPL_pvfk_core + 68
	5   dokibox                             0x0000000100035d06 -[LibraryViewArtistCell observeValueForKeyPath:ofObject:change:context:] + 326
	6   Foundation                          0x00007fff8abfaa03 NSKeyValueNotifyObserver + 382
	7   Foundation                          0x00007fff8abf9bd8 NSKeyValueDidChange + 463
	8   Foundation                          0x00007fff8abfe73d -[NSObject(NSKeyValueObserverNotification) didChangeValueForKey:] + 118
	9   dokibox                             0x000000010002f34f -[LibraryView tableView:viewForTableColumn:row:] + 735
	10  AppKit                              0x00007fff8bb3f480 -[NSTableView(NSTableViewViewBased) makeViewForTableColumn:row:] + 76
	11  AppKit                              0x00007fff8bb3e791 -[NSTableRowData _addViewToRowView:atColumn:row:] + 300
	12  AppKit                              0x00007fff8bb3e4ca -[NSTableRowData _addViewsToRowView:atRow:] + 184
	13  AppKit                              0x00007fff8c2a3f9f -[NSTableRowData _initializeRowView:atRow:] + 373
	14  AppKit                              0x00007fff8bb3b2e4 -[NSTableRowData _addRowViewForVisibleRow:withPriorView:] + 396
	15  AppKit                              0x00007fff8bb3b081 -[NSTableRowData _addRowViewForVisibleRow:withPriorRowIndex:inDictionary:withRowAnimation:] + 254
	16  AppKit                              0x00007fff8bb3a493 -[NSTableRowData _unsafeUpdateVisibleRowEntries] + 1856
	17  AppKit                              0x00007fff8bb39bb7 -[NSTableRowData updateVisibleRowViews] + 230
	18  AppKit                              0x00007fff8bbbeb42 -[NSTableView viewWillDraw] + 178
	19  AppKit                              0x00007fff8bbc8aa1 -[NSView(NSInternal) _sendViewWillDrawAndRecurse:] + 535
	20  AppKit                              0x00007fff8bbc8641 -[NSView(NSLayerKitGlue) _layoutSublayersOfLayer:] + 142
	21  QuartzCore                          0x00007fff93575050 -[CALayer layoutSublayers] + 219
	22  AppKit                              0x00007fff8bbc859d _NSBackingLayerLayoutSublayers + 158
	23  QuartzCore                          0x00007fff93574c34 _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 366
	24  QuartzCore                          0x00007fff93574ab2 _ZN2CA5Layer28layout_and_display_if_neededEPNS_11TransactionE + 24
	25  QuartzCore                          0x00007fff9357424e _ZN2CA7Context18commit_transactionEPNS_11TransactionE + 242
	26  QuartzCore                          0x00007fff93573ea4 _ZN2CA11Transaction6commitEv + 390
	27  QuartzCore                          0x00007fff93584cad _ZN2CA11Transaction17observer_callbackEP19__CFRunLoopObservermPv + 71
	28  CoreFoundation                      0x00007fff945f8da7 __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
	29  CoreFoundation                      0x00007fff945f8d00 __CFRunLoopDoObservers + 368
	30  CoreFoundation                      0x00007fff945ea878 CFRunLoopRunSpecific + 328
	31  HIToolbox                           0x00007fff8908eaef RunCurrentEventLoopInMode + 235
	32  HIToolbox                           0x00007fff8908e86a ReceiveNextEventCommon + 431
	33  HIToolbox                           0x00007fff8908e6ab _BlockUntilNextEventMatchingListInModeWithFilter + 71
	34  AppKit                              0x00007fff8ba99f81 _DPSNextEvent + 964
	35  AppKit                              0x00007fff8ba99730 -[NSApplication nextEventMatchingMask:untilDate:inMode:dequeue:] + 194
	36  AppKit                              0x00007fff8ba8d593 -[NSApplication run] + 594
	37  AppKit                              0x00007fff8ba78a14 NSApplicationMain + 1832
	38  dokibox                             0x0000000100002992 main + 34
	39  dokibox                             0x0000000100001ec4 start + 52
	40  ???                                 0x0000000000000003 0x0 + 3
)
2015-02-08 21:37:35:356 V [main] (Library.m:48) Monitored folder: /Users/mileswu/Music
2015-02-08 21:37:35:358 V [main] (LibraryView.m:162) deleting (null)
2015-02-08 21:37:35:358 V [main] (LibraryView.m:162) deleting (null)
2015-02-08 21:37:35:358 V [main] (LibraryView.m:162) deleting (null)

**Comment by mileswu** [comment]
Fixed in 6838bd60b41053da25d8271759fd936c0cf5b99f.

In particular this was caused by having results displayed that were already deleted.
